govcd/vm.go

```diff
@@ -850,3 +851,186 @@ func (vm *VM) SetGuestCustomizationSection(guestCustomizationSection *types.Gues
 
 	return vm.GetGuestCustomizationSection()
 }
+
+// AddInternalDisk creates disk type *types.DiskSettings to the VM.
+// Returns new disk ID and error.
+// Runs synchronously, VM is ready for another operation after this function returns.
+func (vm *VM) AddInternalDisk(diskData *types.DiskSettings) (string, error) {
+	if vm.VM.HREF == "" {
+		return "", fmt.Errorf("cannot add internal disks - VM HREF is unset")
+	}
+
+	err := vm.validateInternalDiskInput(diskData)
+	if err != nil {
+		return "", err
+	}
+
+	diskSettings := vm.VM.VmSpecSection.DiskSection.DiskSettings
+	if diskSettings == nil {
```

Comment:
Should this function have the "refresh" option or maybe even an implicit  refresh? I suspect that if one didn't refresh the VM before triggering this function - he would wipe all the disk configuration (if he has set the HREF only).

I wonder if it would be safe to assume that if `diskSettings == nil` the VM is actually not refreshed. Wondering if the `diskSettings` structure has any difference when a VM has no disks (but is refreshed) vs being `nil` (not refreshed).

Reply:
I will need to check, forget todo it, though thought about it. A lof things here to lookup after

Reply:
so when where isn't any internal disk, then we get `<DiskSection/>` - code or more less correct. I am just thinking to look for possibility enforce refresh.

Reply:
I added refresh handling. I made it implicit due big risk leave for user and he would forget to pass correct value.